specifyweb/frontend/js_src/lib/components/WorkBench/Status.tsx

```diff
@@ -134,6 +134,44 @@ export function WbStatus({
       </>
     );
 
+  const startTime = React.useRef(Date.now());
+  const [remainingTime, setRemainingTime] = React.useState<number | null>(null);
+
+  // Convert seconds to minutes and seconds and create returned string
+  function formatTime(time: number | null): string {
+    if (time !== null) {
+      const minutes = Math.floor(time / 60);
+      const seconds = time % 60;
+      return `${commonText.timeRemaining()} ${minutes}:${
+        seconds < 10 ? '0' : ''
+      }${seconds}`;
+    }
+    return '';
+  }
+
+  // Calculate the remaining time
+  React.useEffect(() => {
+    if (status.taskstatus === 'PROGRESS') {
+      if (current > 15) {
+        const interval = setInterval(() => {
+          const remainingSeconds = Math.round(
+            (((total - current) / current) * (Date.now() - startTime.current)) /
```

Comment:
(optional) this expression could be split into multiple variables to make it more readable